pos_tare_generate_barcode/static/src/js/screens.js

```diff
@@ -0,0 +1,200 @@
+        next_screen: 'products',
+        previous_screen: 'products',
+        default_tare_value: 0.0,
+        weight_barcode_prefix: null,
+
+        show: function () {
+            this._super();
+            // Fetch the barcode prefix from POS barcode parser rules.
+            this.weight_barcode_prefix = this.get_barcode_prefix();
+            // Setup the proxy
+            var queue = this.pos.proxy_queue;
+            // The pooling of the scale starts here.
+            var self = this;
+            queue.schedule(function () {
+                return self.pos.proxy.scale_read().then(function (weight) {
+                    try {
+                        self.set_tare_weight(weight);
+                    } catch (error) {
+                        var title = _t("Failed to read weight from scale.");
+                        var popup = {title: title, body: error.message};
+                        self.gui.show_popup('error', popup);
+                        self.pos.proxy_queue.clear();
+                    }
+                });
+            }, {duration:150, repeat: true});
+            // Shows a barcode whose weight might be zero, but this is preferred
+            // for UI/UX reasons.
+            this.render_receipt();
+            this.lock_screen(true);
+        },
+        get_barcode_prefix: function () {
+            var barcode_pattern = this.get_barcode_pattern();
+            return barcode_pattern.substr(0, 2);
+        },
+        get_barcode_pattern: function () {
+            var rules = this.get_barcode_rules();
+            var rule = rules.filter(
+                function (r) {
+                    // We select the first (smallest sequence ID) barcode rule
+                    // with the expected type.
+                    return r.type === "tare";
+                })[0];
+            return rule.pattern;
+        },
+        get_barcode_rules: function () {
+            return this.pos.barcode_reader.barcode_parser.nomenclature.rules;
+        },
+        set_tare_weight: function (scale_measure) {
+            var weight = scale_measure.weight;
+            var measure_unit = this.pos.units.filter(
+                function (u) {
+                    return u.name === scale_measure.unit;
+                })[0];
+
+            if (typeof measure_unit === 'undefined') {
+                throw new Error(_.str.sprintf(
+                    _t("The scale sent a measure in %s unit. This unit of "+
+                     "measure (UOM) in not found in the point of sale. You " +
+                     "may need to create a new UOM named %s. The UOM name is "+
+                     "case sensitive."), scale_measure.unit,
+                    scale_measure.unit));
+            }
+
+            if (weight > 0) {
+                var tare_uom = this.pos.config.iface_tare_uom_id[0];
+                var tare_unit = this.pos.units_by_id[tare_uom];
+                this.weight_in_tare_unit = convert_mass(weight,
```

Comment:
@houssine78 The tare UOM is a parameter of the POS. Here we convert the weight expressed in the scale's UOM into the parameterized UOM.